apps/src/templates/textResponses/textReponsesDataApi.js

```diff
@@ -0,0 +1,49 @@
+import PropTypes from 'prop-types';
+
+// Make a request to the server for text responses
+// scriptId is not required; endpoint will use the default script if no scriptId is supplied
+export const loadTextResponsesFromServer = (sectionId, scriptId) => {
+  let requestUrl = `/dashboardapi/section_text_responses/${sectionId}`;
+
+  if (scriptId) {
+    requestUrl += `?script_id=${scriptId}`;
+  }
+
+  return fetch(requestUrl, {
+    credentials: 'same-origin'
+  })
+    .then(response => {
+      return response.json();
+    })
+    .then(responseData => {
+      return convertTextResponseServerData(responseData);
+    });
+};
+
+// Flatten text responses returned from server to remove nested student object
+const convertTextResponseServerData = textResponses => {
```

Comment:
Is this API used anywhere else?  Wondering if we should just change the shape of the response on the back-end so that we don't have to transform on the front-end.

Reply:
I was thinking the same thing, but wanted to limit the scope of my rabbit hole.